### PR TITLE
Add quantum/breast_cancer.py

### PR DESCRIPTION
**Pull request by prakhargurunani**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #2923 `.

### CODE REVIEW

**Comment by prakhargurunani** [comment]
@cclauss Can you also add `hacktoberfest-accepted` label to this PR 😊.

File: quantum/breast_cancer.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import numpy as np
+from sklearn import datasets
+from sklearn.decomposition import PCA
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import MinMaxScaler, StandardScaler
+
+
+def breast_cancer(training_size, test_size, n, plot_data=False):
```
Comment (cclauss) [inline quantum/breast_cancer.py:8]: Needs type hints and doctests as discussed in CONTRIBUTING.md.

File: quantum/breast_cancer.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    training_input = {
+        key: (sample_train[label_train == k, :])[:training_size]
+        for k, key in enumerate(class_labels)
+    }
+    test_input = {
+        key: (sample_test[label_test == k, :])[:test_size]
+        for k, key in enumerate(class_labels)
+    }
+
+    if plot_data:
```
Comment (cclauss) [inline quantum/breast_cancer.py:42]: Algorithmic functions should not print() or plot as discussed in CONTRIBUTING.md.  So this function should return values an allow the caller to catch the exception, print(), and/or plot.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: quantum/breast_cancer.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+from sklearn.preprocessing import MinMaxScaler, StandardScaler
+
+
+def breast_cancer(training_size, test_size, n, plot_data=False):
+    """returns breast cancer dataset
+
+    Wikipedia reference: https://en.m.wikipedia.org/wiki/Breast_cancer
+
+    >>> breast_cancer(10, 4, 7)
+    24.9
```
Comment (cclauss) [inline quantum/breast_cancer.py:14]: The function return four values: `return sample_train, training_input, test_input, class_labels`

**Comment by cclauss** [comment]
Please run `python3 -m doctest -v breast_cancer.py` on your local machine until the tests pass.

**Comment by prakhargurunani** [comment]
> Please run `python3 -m doctest -v breast_cancer.py` on your local machine until the tests pass. This is discussed in CONTRIBUTING.md.

Ok. I will do it.

**Comment by prakhargurunani** [comment]
@cclauss I have accepted your suggestions. Is there anything more to change ?

**Comment by cclauss** [comment]
Closed for ___plagiarism___.
This code is copied from https://qiskit.org/documentation/_modules/qiskit/ml/datasets/breast_cancer.html
